DOC: update release notes with all deprecations for 1.12 release (#19735)
scipy/scipy: doc/source/release/1.12.0-notes.rst

- All wavelet functions have been deprecated, as PyWavelets provides suitable
  implementations; affected functions are: ``signal.{daub, qmf, cascade,
  morlet, morlet2, ricker, cwt}``

> - `scipy.integrate.trapz`, `scipy.integrate.cumtrapz`, and `scipy.integrate.simps` have

Author: ```suggestion
- ``scipy.integrate.trapz``, ``scipy.integrate.cumtrapz``, and ``scipy.integrate.simps`` have
```

  been deprecated in favour of `scipy.integrate.trapezoid`, `scipy.integrate.cumulative_trapezoid`,

>   and `scipy.integrate.simpson` and will be removed in SciPy 1.14.

Author: ```suggestion
  and `scipy.integrate.simpson` respectively and will be removed in SciPy 1.14.
```

> - The deprecation of the ``tol`` argument of ``scipy.sparse.linalg.{gcrotmk,lgmres,minres,tfqmr}``
>   is now consistent with the other iterative solvers and will be removed in SciPy 1.14

Author: I believe this is now correct @h-vetinari 
```suggestion
- The ``tol`` argument of ``scipy.sparse.linalg.{gcrotmk,lgmres,minres,tfqmr,bcg,bicstab,cg,cgs,gmres,qmr}``
  is now deprecated and will be removed in SciPy 1.14
```

Reviewer: Let's pull forward the mention of `rtol` and alphabetize the solvers
```suggestion
- The ``tol`` argument of ``scipy.sparse.linalg.{bcg,bicstab,cg,cgs,gcrotmk,gmres,lgmres,minres,qmr,tfqmr}``
  is now deprecated in favour of ``rtol`` and will be removed in SciPy 1.14.
```

  in favour of ``rtol``. Further, the default value of ``atol`` for these functions is due

>   to change to ``0.0`` in SciPy 1.14.

Reviewer: The deprecation of `tol` happened across the board for all* sparse solvers in 1.12, see #18488.

\* Of course, there's an [exception](https://github.com/scipy/scipy/issues/19704)... 😑

> - The ``tol`` argument of ``scipy.sparse.linalg.{bcg,bicstab,cg,cgs,gcrotmk,gmres,lgmres,minres,qmr,tfqmr}``
>   is now deprecated in favour of ``rtol`` and will be removed in SciPy 1.14.
>   in favour of ``rtol``. Further, the default value of ``atol`` for these functions is due
>   to change to ``0.0`` in SciPy 1.14.

Reviewer: ```suggestion
- The ``tol`` argument of ``scipy.sparse.linalg.{bcg,bicstab,cg,cgs,gcrotmk,gmres,lgmres,minres,qmr,tfqmr}``
  is now deprecated in favour of ``rtol`` and will be removed in SciPy 1.14.
  Furthermore, the default value of ``atol`` for these functions is due
  to change to ``0.0`` in SciPy 1.14.
```

There is an ongoing effort to follow through on long-standing deprecations.
The following previously deprecated features are affected:

- The ``centered`` keyword of `scipy.stats.qmc.LatinHypercube` has been removed.
  Use ``scrambled=False`` instead of ``centered=True``.
- ``scipy.stats.binom_test`` has been removed in favour of `scipy.stats.binomtest`.

> - The use of ``scale='raw'`` has been removed in favour of ``'scale=1'`` in

Author: ```suggestion
- The use of ``scale='raw'`` has been removed in favour of ``scale=1`` in
```

> - The use of ``scale='raw'`` has been removed in favour of ``'scale=1'`` in
>   `scipy.stats.iqr`.

Author: ```suggestion
- In `scipy.stats.iqr`, the use of ``scale='raw'`` has been removed in favour 
  of ``'scale=1'``.
```
